uni/lib/view/library/library.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:percent_indicator/linear_percent_indicator.dart';
+import 'package:tuple/tuple.dart';
+import 'package:uni/model/app_state.dart';
+import 'package:uni/model/entities/library_occupation.dart';
+import 'package:uni/view/common_widgets/page_title.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/general.dart';
+import 'package:uni/view/common_widgets/request_dependent_widget_builder.dart';
+import 'package:uni/view/library/widgets/library_occupation_card.dart';
+
+class LibraryPageView extends StatefulWidget {
+  const LibraryPageView({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => LibraryPageViewState();
+}
+
+class LibraryPageViewState extends GeneralPageViewState<LibraryPageView> {
+  @override
+  Widget getBody(BuildContext context) {
+    return StoreConnector<AppState, Tuple2<LibraryOccupation?, RequestStatus>>(
+        converter: (store) {
+      final LibraryOccupation? occupation =
+          store.state.content['libraryOccupation'];
+      return Tuple2(occupation, store.state.content['libraryOccupationStatus']);
+    }, builder: (context, occupationInfo) {
+      return RequestDependentWidgetBuilder(
+          context: context,
+          status: occupationInfo.item2,
+          contentGenerator: generateOccupationPage,
+          content: occupationInfo.item1,
+          contentChecker: occupationInfo.item2 != RequestStatus.busy,
+          onNullContent: const Center(child: CircularProgressIndicator()));
+    });
+  }
+
+  Widget getFloorRows(LibraryOccupation occupation) {
+    final List<Widget> floors = [];
+    for (int i = 1; i < 7; i += 2) {
+      floors.add(
+          createFloorRow(occupation.getFloor(i), occupation.getFloor(i + 1)));
+    }
+    return Column(
+      children: floors,
+    );
+  }
+
+  Widget generateOccupationPage(occupation, context) {
+    List<Widget> content;
+    content = [
+      const PageTitle(name: 'Biblioteca'),
+      LibraryOccupationCard(),
+    ];
+    if (occupation != null) {
+      content.add(const PageTitle(name: 'Pisos'));
+      content.add(getFloorRows(occupation));
+    }
+
+    return ListView(
+        scrollDirection: Axis.vertical, shrinkWrap: true, children: content);
+  }
```

Comment:
You can inline this for a cleaner look:
```suggestion
    return ListView(
        scrollDirection: Axis.vertical, shrinkWrap: true, children: [
               const PageTitle(name: 'Biblioteca'),
               LibraryOccupationCard(),
               if (occupation != null) const PageTitle(name: 'Pisos'),
               if (occupation != null) getFloorRows(occupation),
        ]);
  }
```